osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -42,7 +48,21 @@ protected override double StrainValueOf(DifficultyHitObject current)
             var osuCurrent = (OsuDifficultyHitObject)current;
 
             double distance = Math.Min(single_spacing_threshold, osuCurrent.TravelDistance + osuCurrent.JumpDistance);
-            double deltaTime = Math.Max(max_speed_bonus, current.DeltaTime);
+            double deltaTime = current.DeltaTime;
+
+            // Aim to nerf cheesy rhythms (Very fast consecutive doubles with large deltatimes between)
+            double deltaTimeThreshold = greatWindow * 2;
+
+            if (Previous.Count > 0 && deltaTime < deltaTimeThreshold && Previous[0].DeltaTime > deltaTime)
+            {
+                double speedWindowRatio = deltaTime / deltaTimeThreshold;
+                deltaTime = Interpolation.Lerp(Previous[0].DeltaTime, deltaTime, speedWindowRatio);
+            }
+
+            // Cap deltatime to the OD 300 hitwindow.
+            // 0.93 is derived from making sure 260bpm OD8 streams aren't nerfed harshly
+            var hitWindowNerf = deltaTime / (greatWindow * 2 * 0.93);
+            deltaTime /= Math.Clamp(hitWindowNerf, 0.92, 1);
```

Comment:
`0.92`?

Reply:
Unless I'm misunderstanding, could this whole thing just be simplified to:
```cs
deltaTime = Math.Clamp(greatWindow * 2 * 0.93, deltaTime, deltaTime / 0.92)
```
The purpose of using a separate variable `hitWindowNerf` isn't clear to me

Reply:
Yeah, that's a value that was worked out through testing, which is just just limiting the effect of the nerf. Do you think that could do with a comment?

Reply:
It's split up like that for the sake of readibility, I definitely think the simplified form harms the comprehension (unless that doesn't really matter)

Reply:
In my opinion it's harder to understand when it's split up like that, your comment says "`Cap deltatime to the OD 300 hitwindow.`" yet it's not obvious how `hitWindowNerf` does that. I think this could wait for other opinions

Reply:
Ah, when you put it like that, `hitWindowNerf` is certainly the wrong name since it's meant to serve as the ratio of `deltaTime` to the hit window (slightly smaller as the comment states hence the 0.93). Would renaming the variable make it more clear here, or do you think the problem is in the splitting of the lines?

Reply:
I don't have a more appropriate name but if you can think of one I think that would work

Reply:
I've moved some stuff around, should be much clearer now. Even at a single line now whilst remaining completely readable

Reply:
Nice, also yeah I think you should add a comment explaining that `0.92` value (assuming it has a different purpose to dividing `speedWindowRatio` by `0.93`)

Reply:
Added, should be fine now.